inference-engine/tests/functional/inference_engine/ngraph_reshape_tests.cpp

```diff
@@ -141,6 +141,36 @@ TEST_F(NGraphReshapeTests, ReshapeSpatialReLU) {
     ASSERT_EQ(ngraph->get_results()[0]->get_shape(), ngraph::Shape({1, 3, 25, 25}));
 }
 
+TEST_F(NGraphReshapeTests, ReshapeSpatialReLUWithoutReplaceParameter) {
+    std::shared_ptr<ngraph::Function> ngraph;
+    {
+        ngraph::PartialShape shape({1, 3, 22, 22});
+        ngraph::element::Type type(ngraph::element::Type_t::f32);
+        auto param = std::make_shared<ngraph::op::Parameter>(type, shape);
+        auto relu = std::make_shared<ngraph::op::Relu>(param);
+        auto result = std::make_shared<ngraph::op::Result>(relu);
+
+        ngraph::ParameterVector params = {param};
+        ngraph::ResultVector results = {result};
+
+        ngraph = std::make_shared<ngraph::Function>(results, params);
+    }
+
+    ASSERT_EQ(ngraph->get_parameters()[0]->get_shape(), ngraph::Shape({1, 3, 22, 22}));
+    ASSERT_EQ(ngraph->get_results()[0]->get_shape(), ngraph::Shape({1, 3, 22, 22}));
+
+    {
+        auto& shape = ngraph->get_parameters()[0]->get_partial_shape();
+        shape = {1, 3, 25, 25};
```

Comment:
looks very tricky.. Maybe we need to remove copy of `PartialShape`? instead of & to it.

@slyalin how do you think?

Reply:
I think we can introduce the new method, but in any case we cannot change the current  API without backward compatibility issues.

Reply:
In new API we can :)

Reply:
or, just add a comment :)

Reply:
I propose to resolve it in the separate PR.

Reply:
Changed to set_partial_shape